sim-lib/src/lib.rs

```diff
@@ -878,46 +883,51 @@ async fn write_payment_results(
     mut receiver: Receiver<(Payment, PaymentResult)>,
     listener: Listener,
     print_batch_size: u32,
+    no_results: bool,
 ) -> Result<(), SimulationError> {
-    let mut writer = WriterBuilder::new().from_path(format!(
-        "simulation_{:?}.csv",
-        SystemTime::now()
-            .duration_since(SystemTime::UNIX_EPOCH)
-            .unwrap()
-            .as_secs()
-    ))?;
+    let mut writer = if !no_results {
+        Some(WriterBuilder::new().from_path(format!(
+            "simulation_{:?}.csv",
+            SystemTime::now()
+                .duration_since(SystemTime::UNIX_EPOCH)
+                .unwrap()
+                .as_secs()
+        ))?)
+    } else {
+        None
+    };
 
     let mut result_logger = PaymentResultLogger::new();
-
-    let mut counter = 1;
+    let mut counter = 0;
     loop {
         tokio::select! {
             biased;
             _ = listener.clone() => {
                 log::debug!("Simulation results consumer received shutdown signal.");
-                break writer.flush().map_err(|_| SimulationError::FileError)
+                return writer.map(|ref mut w| w.flush().map_err(|_| SimulationError::FileError)).unwrap_or(Ok(()));
```

Comment:
Not sure if this is canonical, but is `map_or` a good fit here? 
`writer.map_or(Ok(()), |mut w| w.flush().map_err(|_| SimulationError::FileError))`

Non-blocking, just 🦀 nit.

Reply:
I think it is!

Reply:
My first rust nit 🦀 😭

Reply:
🎉